pkg/controllers/manifestwork.go

```diff
@@ -100,17 +100,52 @@ func ScaffoldManifestwork(hyd *hypdeployment.HypershiftDeployment) (*workv1.Mani
 	return w, nil
 }
 
-func setManifestWorkSelectivelyDeleteOption(mw *workv1.ManifestWork, hostingNamespace string) {
-	mw.Spec.DeleteOption = &workv1.DeleteOption{
-		PropagationPolicy: workv1.DeletePropagationPolicyTypeSelectivelyOrphan,
-		SelectivelyOrphan: &workv1.SelectivelyOrphan{
-			OrphaningRules: []workv1.OrphaningRule{
-				{
-					Resource: "namespaces",
-					Name:     hostingNamespace,
+func setManifestWorkSelectivelyDeleteOption(mw *workv1.ManifestWork, hyd *hypdeployment.HypershiftDeployment) {
+	hostingNamespace := helper.GetHostingNamespace(hyd)
+
+	if hyd.Spec.Override == hypdeployment.InfraOverrideDestroy {
+		mw.Spec.DeleteOption = &workv1.DeleteOption{
+			PropagationPolicy: workv1.DeletePropagationPolicyTypeSelectivelyOrphan,
+			SelectivelyOrphan: &workv1.SelectivelyOrphan{
+				OrphaningRules: []workv1.OrphaningRule{
+					{
+						Resource: "namespaces",
+						Name:     hostingNamespace,
+					},
+					{
+						Group:     "hypershift.openshift.io",
+						Resource:  "hostedclusters",
+						Name:      hyd.Name,
+						Namespace: hostingNamespace,
+					},
+					{
+						Group:     "hypershift.openshift.io",
+						Resource:  "nodepools",
+						Name:      hyd.Name,
+						Namespace: hostingNamespace,
+					},
 				},
 			},
-		},
+		}
+	} else if hyd.Spec.Override == hypdeployment.DeleteHostingNamespace {
+		mw.Spec.DeleteOption = &workv1.DeleteOption{
```

Comment:
Setting the deleteOption to `DeletePropagationPolicyTypeForeground` will delete all resources including the ns.